crates/network/src/transaction/builder.rs

```diff
@@ -350,4 +361,39 @@ pub trait TransactionBuilder<N: Network>: Default + Sized + Send + Sync + 'stati
         self,
         wallet: &W,
     ) -> impl_future!(<Output = Result<N::TxEnvelope, TransactionBuilderError<N>>>);
+
+    /// Set the input data for the transaction, respecting the input kind
+    fn set_input_respecting_kind<T: Into<Bytes>>(&mut self, input: T, kind: TransactionInputKind) {
```

Comment:
I think we can shorten this
```suggestion
    fn set_input_kind<T: Into<Bytes>>(&mut self, input: T, kind: TransactionInputKind) {
```